Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -102,6 +120,8 @@ protected void CopyCommonSettings<TOther>(ClientBuilderBase<TOther> source)
             TokenAccessMethod = source.TokenAccessMethod;
             CallInvoker = source.CallInvoker;
             UserAgent = source.UserAgent;
+            GrpcImplementation = source.GrpcImplementation;
+            EmulatorDetection = source.EmulatorDetection;
```

Comment:
This might not be copying from a builder of the same client, not sure if it makes sense to copy `EmulatorDetection`. Imagine `TOther` supports emulator and `TClient` doesn't, and we copy `EmulatorOnly` to this builder which is for `TClient`. The emulator for `TClient` will never be set and in `TClientClientBuilder` the `EmulatorDetection` property will be `protected` so it can't even be changed back. Calling `Build` will always fail.

Reply:
Yes, that's true. Hmm. Maybe we need to not do that, but document it very clearly. Tricky.